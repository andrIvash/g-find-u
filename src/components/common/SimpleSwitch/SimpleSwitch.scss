.simpleSwitch {
    display: flex;
    align-items: center;
}

.simpleSwitch__label{
    display: flex;
    align-items: center;
}