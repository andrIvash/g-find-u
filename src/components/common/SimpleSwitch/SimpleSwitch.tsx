import * as React from 'react';
import Switch from '@mui/material/Switch';
import { Props } from "./index";
import './SimpleSwitch.scss';

export const SimpleSwitch = (props: Props) => {
    
    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        props.onChange(event.target.checked)
    };

    return (
        <div className='simpleSwitch'>
            { props.leftLabel ? (
                <div className='simpleSwitch__label simpleSwitch__leftLabel'>
                    {props.leftLabel}
                </div>
            ) : undefined }
            <Switch
                checked={props.checked}
                onChange={handleChange}
                inputProps={{ 'aria-label': 'controlled' }}
                color="default"
            />
            { props.rightLabel ? (
                <div className='simpleSwitch__label simpleSwitch__rightLabel'>
                    {props.rightLabel}
                </div>
            ) : undefined }
        </div>
    )
}
