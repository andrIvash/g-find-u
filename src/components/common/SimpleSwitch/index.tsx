import { SimpleSwitch } from './SimpleSwitch';

export interface Props {
    checked: boolean,
    onChange: (value?: boolean) => void,
    leftLabel?: string | React.ReactNode,
    rightLabel?: string | React.ReactNode,
    className?: string
}

export default SimpleSwitch;