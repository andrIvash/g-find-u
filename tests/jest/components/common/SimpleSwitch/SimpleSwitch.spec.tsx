import React from 'react';
import '@testing-library/jest-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import SimpleSwitch from '../../../../../src/components/common/SimpleSwitch';

describe('SimpleSwitch Component', () => {
  const mockOnChange = jest.fn();

  it('renders correctly', () => {
    const { asFragment } = render(<SimpleSwitch onChange={mockOnChange} checked={false} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays left and right labels when provided', () => {
    render(<SimpleSwitch onChange={mockOnChange} leftLabel="Left" rightLabel="Right" checked={false} />);
    expect(screen.getByText('Left')).toBeInTheDocument();
    expect(screen.getByText('Right')).toBeInTheDocument();
  });

  it('toggles switch state on click', () => {
    render(<SimpleSwitch onChange={mockOnChange} checked={false} />);
    const switchControl = screen.getByRole('checkbox');
    fireEvent.click(switchControl);
    expect(mockOnChange).toHaveBeenCalledWith(true);
  });
});