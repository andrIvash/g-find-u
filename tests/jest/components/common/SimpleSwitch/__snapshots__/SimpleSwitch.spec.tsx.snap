// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleSwitch Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="simpleSwitch "
    data-testid="SimpleSwitch"
  >
    <span
      class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
    >
      <span
        class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorDefault PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorDefault css-1i8lvs-MuiButtonBase-root-MuiSwitch-switchBase"
      >
        <input
          aria-label="controlled"
          class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
          type="checkbox"
        />
        <span
          class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
      />
    </span>
  </div>
</DocumentFragment>
`;
